### ISSUE TITLE
Accessibility: Button incorrectly given the role of link

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#52147]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

Buttons incorrectly have <a> tags on them, though they do not function as links and are not redirecting to another URL. Buttons are elements that trigger an action on a page (opening a lightbox, expanding an accordion, submitting a form, closing a modal).

![Image](https://github.com/user-attachments/assets/10c6296e-62c6-4397-b7d1-4e6ca11dac62)

### Component or Page: 

WooCommerce Store Notice

### Name affected element:

Dismiss button

### Relevant code snippets:

`<a href="#" class="woocommerce-store-notice__dismiss-link">Dismiss</a>`

### Recommended fix:

Use the appropriate <button> tag on the element.
 
 If you cannot replace a <a> tag with a button, it is possible to remediate a link to function as a button with ARIA. The link would need to be given a button role and href=""#"" to ensure that it can be focused with the keyboard. You would also need to write JavaScript to ensure that the link will function properly as a button for keyboard users and can be triggered with the space bar in addition to the return/enter key. For more information on turning a link into a button, see: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role
